src/app/shared/components/header/header.component.ts

```diff
@@ -12,5 +17,47 @@ export class HeaderComponent {
     searchValue: [''],
   });
 
-  constructor(private fb: FormBuilder) {}
+  constructor(
+    private fb: FormBuilder,
+    private apiService: ApiService,
+    private paginationService: PaginationService,
+    private moviesService: MoviesService,
+  ) {}
+
+  public search(): void {
+    const requestQuery: string = this.searchForm.value.searchValue;
+
+    if (requestQuery.length === 0) {
```

Comment:
you have a lot of checks for zero. I recommend making a util function that checks for zero and use it in such places